src/handlers/records-query.ts

```diff
@@ -216,20 +230,26 @@ export class RecordsQueryHandler implements MethodHandler {
   }
 
   /**
-   * Does the recordQuery filter explicitly for published records.
-   * Checks that there is either an explicit datePublished filter, or published is explicitly not set to false.
+   * Checks if the recordQuery filter supports returning published records.
    */
   private static filtersForPublishedRecords(recordsQuery: RecordsQuery): boolean {
     const { filter } = recordsQuery.message.descriptor;
-    return filter.datePublished !== undefined || filter.published !== false;
+    // When `published` and `datePublished` range are both undefined, published records can be returned.
+    if (filter.datePublished === undefined && filter.published === undefined) {
+      return true;
+    }
+    return filter.datePublished !== undefined || filter.published === true;
   }
 
   /**
-   * Does the recordQuery filter explicitly for unpublished records.
-   * Checks that published is not explicitly set to true, and datePublished is explicitly undefined.
+   * Checks if the recordQuery filter supports returning unpublished records.
    */
   private static filtersForUnpublishedRecords(recordsQuery: RecordsQuery): boolean {
     const { filter } = recordsQuery.message.descriptor;
-    return filter.published !== true && filter.datePublished === undefined;
+    // When `published` and `datePublished` range are both undefined, unpublished records can be returned.
+    if (filter.datePublished === undefined && filter.published === undefined) {
+      return true;
+    }
+    return filter.published === false && filter.datePublished === undefined;
```

Comment:
Sanity: would this suffice

```suggestion
    return filter.published !== true || filter.datePublished === undefined;
```